os/arch/arm/src/amebasmart/amebasmart_smp.c

```diff
@@ -73,6 +74,79 @@ void rtk_core1_power_off(void)
 	HAL_WRITE32(SYSTEM_CTRL_BASE_HP, REG_HSYS_HP_PWC, val);
 }
 
+void vPortSecondaryStart(void)
+{
+        int count = 10;
+        int state;
+
+        /* take the lock, as we are about to put cpu1 idle task back active */
+        sched_lock();
+
+        /* turn on CPU1's power domain, require time to stabilize */
+        rtk_core1_power_on();
+        DelayMs(1);
+
+        /* renable scu and invalidate tags on all processors */
+        arm_enable_smp(0);
+
+        /* cold-boot the core */
+        lldbg("Booting secondary core... \n");
+        BaseType_t err = psci_cpu_on(SECONDARY_CORE_ID, (unsigned long)__cpu1_start);
+        DEBUGASSERT(err >= 0);
+
+        /* await for PSCI to report cpu state */
+        do {
+                state  = psci_affinity_info(1, 0);
+                if (state == 1) {
+                        break;
+                }
+
+                DelayUs(50);
+        } while (count--);
+
+        lldbg("Secondary core booted?... %d\n", err);
+        cpu_set_state(SECONDARY_CORE_ID, CPU_RUNNING);
+
+        /* fire SGI1 to kick the freshly booted CPU into TizenRT Idle Task */
+        os_smp_start();
```

Comment:
Could you please check the sequence cpu on/off and seperate & mapping HAL layer?

it looks,

CPU ON
1. rtk_core1_power_on : control HW power block
2. arm_enable_smp :  Enable the SCU and make certain that current CPU is participating in the SMP cache coherency.
3. PSCI CPU ON -> verify psci cpu state
4. os_smp_start -> up_cpu_start (all core) -> SGI1

CPU OFF
1. up_cpu_hatplug -> SGI3
2. PSCI CPU OFF -> verify psci cpu state
3. rtk_core1_power_off : control HW power block

1) we need to control each CPU, cpu1 cpu2 cpu3
2) we will make PSCI layer in armv7-a layer. Please consider it